models/ape_store/base/ape_store_base_trades.sql

```diff
@@ -0,0 +1,104 @@
+{{ config(
+    alias = 'trades',
+    schema = 'ape_store_base',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['blockchain', 'tx_hash', 'tx_index'],
+    post_hook = '{{ expose_spells(
+                        blockchains = \'["base"]\',
+                        spell_type = "project",
+                        spell_name = "ape_store",
+                        contributors = \'["whale_hunter"]\') }}'
+   )
+}}
+
+{% set project_start_date = '2024-04-04' %}
+{% set blockchain = 'base' %}
+{% set weth_contract_address = '0x4200000000000000000000000000000000000006' %}
+
+with
+    deployments as (
```

Comment:
since this CTE can't be filtered incrementally (assumption), i wonder if we take a similar approach to previous spells and make this it's own spell that is incremental, then read from it here. this only applies if we expect the result set here to continue to grow (especially if exponentially). do you foresee that happening?

Reply:
to be clear:
this question is related to the solana bot trades spells where we had to move initial CTEs to their own spells upstream rather than the CTE within due to performance constraints. i'm wondering if this spell fits a similar pattern and to get ahead of it. base is much more performant than solana, but just curious